src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -1102,6 +1105,71 @@ TEST_F(ClusterTaskManagerTest, TestSpillWaitingTasks) {
   AssertNoLeaks();
 }
 
+TEST_F(ClusterTaskManagerTest, PinnedArgsMemoryTest) {
+  /*
+    Total memory required by executing tasks' args stays under the specified
+    threshold. A single task whose total arguments is greater than the
+    threshold will not starve.
+  */
+  std::shared_ptr<MockWorker> worker =
+      std::make_shared<MockWorker>(WorkerID::FromRandom(), 1234);
+  std::shared_ptr<MockWorker> worker2 =
+      std::make_shared<MockWorker>(WorkerID::FromRandom(), 12345);
+  pool_.PushWorker(std::static_pointer_cast<WorkerInterface>(worker2));
+  pool_.PushWorker(std::static_pointer_cast<WorkerInterface>(worker));
+
+  rpc::RequestWorkerLeaseReply reply;
+  int num_callbacks = 0;
+  int *num_callbacks_ptr = &num_callbacks;
+  auto callback = [num_callbacks_ptr](Status, std::function<void()>,
+                                      std::function<void()>) {
+    (*num_callbacks_ptr) = *num_callbacks_ptr + 1;
+  };
+
+  /* This task can run. */
+  default_arg_size_ = 600;
+  auto task = CreateTask({{ray::kCPU_ResourceLabel, 1}}, 1);
+  task_manager_.QueueAndScheduleTask(task, &reply, callback);
+  ASSERT_EQ(num_callbacks, 1);
+  ASSERT_EQ(leased_workers_.size(), 1);
+  ASSERT_EQ(pool_.workers.size(), 1);
+  AssertPinnedTaskArgumentsPresent(task);
+
+  /* This task cannot run because it would put us over the memory threshold. */
```

Comment:
Can you also add a test that another task can run if it shares the same input arg with the first task?